### PR TITLE
Removing classical and quantum registers from to/from qiskit function…

### PR DESCRIPTION
**Pull request by simonwa7**
…s. All qubits will be added to the same quantum register

### CODE REVIEW

File: src/python/zquantum/core/circuit/_gate.py [hunk 1]
```diff
@@ -23,18 +23,18 @@
...
             for quantum computation, as described in 'label'.
         qubits: list[Qubit]
-            List of core.qubit.Qubit object that the gate operates on. 
+            List of core.qubit.Qubit object that the gate operates on.
         params: list[any]
-            List of parameters associated with the gate. 
+            List of parameters associated with the gate.
         The above three attributes follow a standard convention that is enforced throughout
-        Zap OS. 
+        Zap OS.
```
Comment (mstechly) [inline src/python/zquantum/core/circuit/_gate.py:37]: I think we can drop "Zap OS" naming from here ;)

File: src/python/zquantum/core/circuit/_gate.py [hunk 2]
```diff
@@ -377,7 +386,7 @@ def to_cirq(self, input_cirq_qubits=None):
         if self.name == "ZZ":
             return cirq.ZZ(cirq_qubits[0], cirq_qubits[1]) ** (params[0] * 2 / pi)
 
-    def to_qiskit(self, qreg=None):
+    def to_qiskit(self, qreg, creg):
```
Comment (mstechly) [inline src/python/zquantum/core/circuit/_gate.py:389]: Shouldn't `creg` be an optional parameter that we should create as some default if it's not provided?

File: src/python/zquantum/core/utils_test.py [hunk 3]
```diff
@@ -304,19 +308,10 @@ def test_two_value_estimates_are_equal_iff_their_values_and_precisions_are_equal
 
 
 @pytest.mark.parametrize(
-    "estimate",
-    [
-        ValueEstimate(2.0),
-        ValueEstimate(5.0, precision=1e5)
-    ]
+    "estimate", [ValueEstimate(2.0), ValueEstimate(5.0, precision=1e5)]
```
Comment (mstechly) [inline src/python/zquantum/core/utils_test.py:311]: It doesn't seem to really matter, but precision here is `1e5`, shouldn't it be `1e-5`?

**Review by mstechly — APPROVED** [review]

File: src/python/zquantum/core/circuit/_gate.py (hunk 2 above)
Comment (simonwa7) [reply src/python/zquantum/core/circuit/_gate.py:389]: I was toying with this idea, but I don't think it makes sense to create registers that are only "seen" by the Gate class. The register should be created/controlled by the circuit itself.

File: src/python/zquantum/core/circuit/_gate.py (hunk 1 above)
Comment (simonwa7) [reply src/python/zquantum/core/circuit/_gate.py]: bc7736d

File: src/python/zquantum/core/utils_test.py (hunk 3 above)
Comment (simonwa7) [reply src/python/zquantum/core/utils_test.py]: ccf2ad5

File: src/python/zquantum/core/circuit/_gate.py (hunk 2 above)
Comment (mstechly) [reply src/python/zquantum/core/circuit/_gate.py:389]: Ok, sounds good to me :)

**Pull request merged**